src/java.base/share/classes/java/util/zip/ZipFile.java

```diff
@@ -123,11 +123,14 @@ public class ZipFile implements ZipConstants, Closeable {
     public static final int OPEN_DELETE = 0x4;
 
     /**
-     * Flag which specifies whether the validation of the Zip64 extra
-     * fields should be disabled
+     * Flag to specify whether the Extra ZIP64 validation should be
+     * disabled.
      */
-    private static final boolean disableZip64ExtraFieldValidation =
-            GetBooleanAction.privilegedGetProperty("jdk.util.zip.disableZip64ExtraFieldValidation");
+    private static final boolean DISABLE_ZIP64_EXTRA_VALIDATION =
+            GetPropertyAction.privilegedGetProperty(
+                    "jdk.util.zip.disableZip64ExtraFieldValidation", "false")
+                    .equalsIgnoreCase("true");
```

Comment:
This doesn't quite work for -Djdk.util.zip.disableZip64ExtraFieldValidation, this is closer to what you want here:
```
    private static final boolean DISABLE_ZIP64_EXTRA_VALIDATION;
    static {
        String value = GetPropertyAction.privilegedGetProperty(
                "jdk.util.zip.disableZip64ExtraFieldValidation");
        if (value == null) {
            DISABLE_ZIP64_EXTRA_VALIDATION = false;
        } else {
            DISABLE_ZIP64_EXTRA_VALIDATION = value.isEmpty() || value.equalsIgnoreCase("true");
        }
    }
```

Reply:
Reworked similar to above.

Reply:
> Reworked similar to above.

Thanks. Related is that there aren't tests in this PR that run with -Djdk.util.zip.disableZip64ExtraFieldValidation and -D"jdk.util.zip.disableZip64ExtraFieldValidation=true to check that zip implementations won't reject the zip/JAR files with wonky extra fields. Is this something for another PR?

Reply:
I can look to add the additional proposed test separately under an additional PR